### PR TITLE
First commit.

### PR DESCRIPTION
**Pull request by HKV203**
Using PostgreSQL, JDBC, Hikari. Decided not to jump high at first and added only the bookstore books constant table. Still don't know what to do with books lists in all the orders to fit them in a table.

Liquibase is connected as plugin in Maven and generates log files throw command line.

### CODE REVIEW

File: src/main/java/com/andersen/bookstore/controller/Controller.java [hunk 1]
```diff
@@ -14,44 +14,54 @@ public class Controller {
     private final DataControl dataControl;
     private WebView webView;
     private String filepath;
+    private Bookstore bookstore;
+    private static Properties properties;
```
Comment (Alexmansar) [inline src/main/java/com/andersen/bookstore/controller/Controller.java]: why static?

File: src/main/java/com/andersen/bookstore/controller/DataControl.java [hunk 2]
```diff
@@ -52,4 +58,23 @@ public void saveLastState(Bookstore bookstore, String filepath) {
             System.out.println(Arrays.toString(e.getStackTrace()));
         }
     }
+
+    public List<Book> fetchBooks() {
+        String SQL_QUERY = "select * from library";
```
Comment (Alexmansar) [inline src/main/java/com/andersen/bookstore/controller/DataControl.java]: use like a constant. I mean on the class layer

File: src/main/java/com/andersen/bookstore/controller/HikariCPDataSource.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class HikariCPDataSource {
+
+    private static final HikariConfig config = new HikariConfig();
+    private static final HikariDataSource ds;
+    private static final Properties properties = Controller.getProperties();
```
Comment (Alexmansar) [inline src/main/java/com/andersen/bookstore/controller/HikariCPDataSource.java]: if irs static filed, use UPPER_CASE

**Comment by Alexmansar** [comment]
> Using PostgreSQL, JDBC, Hikari. Decided not to jump high at first and added only the bookstore books constant table. Still don't know what to do with books lists in all the orders to fit them in a table.
> 
> Liquibase is connected as plugin in Maven and generates log files throw command line.

You can also use Liquibase to add some kind of book at application startup.

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@Mopgonpblr Hi)) I add some comments, please check them

**Comment by HKV203** [comment]
@Alexmansar I didn't find much about what I can do with Liquibase without using Spring/Hibernate.

File: src/main/resources/Data.txt
Comment (Alexmansar) [inline src/main/resources/Data.txt:1]: why txt not json?

**Comment by Alexmansar** [comment]
@Mopgonpblr Chech ypur CI)

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@Mopgonpblr And small question

**Comment by HKV203** [comment]
> @Mopgonpblr Chech ypur CI)

@Alexmansar that's what I've been doing with test commits. Git Action refuses to connect to my database even after I've changed postgresql.conf variable "listen_addresses" to "*". My local tests are successful

File: src/main/resources/Data.txt
Comment (HKV203) [reply src/main/resources/Data.txt:1]: txt worked for me since we started working with data reading. I didn't feel the need to change it to JSON

**Comment by Alexmansar** [comment]
> > @Mopgonpblr Chech ypur CI)
> 
> @Alexmansar that's what I've been doing with test commits. Git Action refuses to connect to my database even after I've changed postgresql.conf variable "listen_addresses" to "*". My local tests are successful

But why you need it?

**Comment by Alexmansar** [comment]
DataControlTest your problem test
use mocks here, not real objects

**Comment by HKV203** [comment]
> DataControlTest your problem test use mocks here, not real objects

@Alexmansar  I guess it's late for me. 😅 
I found out that the reason was probably that I turned my Hikari data source into an object

**Review by Alexmansar — APPROVED** [review]
@Mopgonpblr Hi) please, merge it)

**Pull request merged**